p42/src/bank.rs

```diff
@@ -0,0 +1,154 @@
+    pub fn get_user_mut(&mut self, name: &str) -> Option<&mut User> {
+        self.users.get_mut(name)
+    }
+
+    /// Gets the total assets and total liabilities for the bank.
+    pub fn calc_balance(&self) -> (i64, i64) {
+        let mut total_liabilities = 0i64;
+        let mut total_assets = 0i64;
+
+        for user in self.users.values() {
+            if user.balance < 0 {
+                total_assets = total_assets
+                    .checked_add(user.balance)
+                    .ok_or("Overflow in total assets")
+                    .unwrap();
+            } else {
+                total_liabilities = total_liabilities
+                    .checked_add(user.balance)
+                    .ok_or("Overflow in total liabilities")
+                    .unwrap();
+            }
+        }
+
+        (total_liabilities, total_assets)
+    }
+
+    /// Transfers amount from one user to another.
+    pub fn transfer_funds(
+        &mut self,
+        from_user: &str,
+        to_user: &str,
+        amount: u64,
+    ) -> Result<(), String> {
+        let from = self.get_user(from_user).ok_or("From user not found")?;
+
+        if from.balance + i64::try_from(from.credit_line).expect("Unable to convert balance to i64")
+            < i64::try_from(amount).expect("Unable to convert amount to i64")
+        {
+            return Err("Insufficient credit limit".to_string());
+        }
+        let from_balance = from
+            .balance
+            .checked_sub(i64::try_from(amount).expect("Unable to convert amount to i64"))
+            .ok_or("Underflow in balance")
+            .unwrap();
+
+        let to_mut = self.get_user_mut(to_user).ok_or("To user not found")?;
+        let to_balance = to_mut
+            .balance
+            .checked_add(i64::try_from(amount).expect("Unable to convert amount to i64"))
+            .ok_or("Overflow in balance")
+            .unwrap();
+        to_mut.balance = to_balance;
+
+        let from_mut = self
+            .get_user_mut(from_user)
+            .expect("This should not be reached, because this user was checked earlier");
+        from_mut.balance = from_balance;
+
+        Ok(())
+    }
+
+    /// Accrues interest on the user balances.
+    pub fn accrue_interest(&mut self) {
+        let credit_interest = self.credit_interest;
+        let debit_interest = self.debit_interest;
+        for user in self.users.values_mut() {
+            let interest = if user.balance < 0 {
+                credit_interest
+            } else {
+                debit_interest
+            };
+            let new_balance = user
+                .balance
+                .checked_add(
+                    user.balance
+                        * i64::try_from(interest).expect("Unable to convert interest to i64")
+                        / 10000,
+                )
+                .ok_or("Overflow/Underflow in balance")
+                .unwrap();
```

Comment:
You can just use `expect("Overflow/Underflow in balance")` instead of `ok_or(..).unwrap()`.